yaksh/models.py

```diff
@@ -2086,6 +2086,21 @@ def is_attempt_inprogress(self):
     def get_previous_answers(self, question):
         return self.answers.filter(question=question).order_by('-id')
 
+    def get_question_by_index(self, all_ordered_questions, questions_by_type):
```

Comment:
Also, ```all_ordered_questions``` need not be passed as an argument, since it's an answer paper model method you can get it for each instance using ```self.get_all_ordered_questions```